### PR TITLE
Release 1.4.1

### PR DESCRIPTION
**Pull request by gouttegd**
This PR updates the 1.4.x maintenance branch to:

- update ROBOT to the newest version 1.9.4;
- update all Python packages to reflect the current state of the master branch;
- prepare the branch for a 1.4.1 release.

### CODE REVIEW

**Review by matentzn — APPROVED** [review]
For a moment I thought I could sneak the latest constraints.txt in but there is one oddity there that I want to check first. Lets go ahead with this!

**Comment by matentzn** [comment]
How does this work now? do I switch to this branch and make a multiarch release as usual? What will happen to this branch afterwards?

**Comment by gouttegd** [comment]
Well, first, I need to sort out whatever happened here: https://github.com/INCATools/ontology-development-kit/actions/runs/5088547393/jobs/9145088129#step:6:9894

Looks like a Python package is (again) misbehaving, but what is weird is that no problem occurred when the tests ran against the master branch last night.

Once that is sorted and this PR is merged in `BRANCH-1.4-MAINTENANCE`, then you can checkout that branch and start the build process as usual.

What will happen to that branch depends on whether we’ll need a 1.4.2 release (e.g. if another ROBOT version becomes available before we’re ready to publish ODK 1.5) or not: if we do, then whatever changes needed to make 1.4.2 will continue happening on that branch. If we don’t, then the branch dies off and we’ll start a `BRANCH-1.5-MAINTENANCE` branch just after 1.5 is released.

**Comment by gouttegd** [comment]
OK, need to cherry-pick the fixes to the test suite that happened shortly after the 1.4 release, then it should be good.

File: seed-via-docker.sh [hunk 1]
```diff
@@ -9,4 +9,4 @@ ODK_TAG=${ODK_TAG:-latest}
 ODK_GITNAME=${ODK_GITNAME:-$(git config --get user.name)}
 ODK_GITEMAIL=${ODK_GITEMAIL:-$(git config --get user.email)}
 
-docker run -u $(id -u):$(id -g) -v $PWD:/work -w /work --rm obolibrary/$ODK_IMAGE:$ODK_TAG /tools/odk.py seed --gitname "$ODK_GITNAME" --gitemail "$ODK_GITEMAIL" "$@"
```
Comment (matentzn) [inline seed-via-docker.sh:12]: Intended change?

**Comment by gouttegd** [comment]
[All tests pass](https://github.com/INCATools/ontology-development-kit/actions/runs/5091204840/jobs/9150928275) with the latest cherry picks, so merging to the 1.4-maintenance branch now.

Up to you to to start building the release. I’ll wait until the images are available on the Docker Hub before tagging the release here on GitHub.

**Pull request merged**